models/labels/addresses/__single_category_labels__/balancer_v2/labels_balancer_v2_pools_arbitrum.sql

```diff
@@ -1,243 +1,284 @@
-        AND SUBSTRING(c.poolId, 0, 42) = cc.output_0
-        {% if is_incremental() %}
-        WHERE c.evt_block_time >= date_trunc("day", now() - interval '1 week')
-          AND cc.call_block_time >= date_trunc("day", now() - interval '1 week')
-        {% endif %}
-    )
-
-    UNION ALL
-
-    SELECT pool_id,
-           zip.tokens AS token_address,
-           zip.weights / pow(10, 18) AS normalized_weight,
-           symbol,
-           pool_type
-    FROM (
-        SELECT c.poolId AS pool_id,
-               explode(arrays_zip(cc.tokens, cc.weights)) AS zip,
-               cc.symbol,
-               'IP' AS pool_type
-        FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} c
-        INNER JOIN {{ source('balancer_v2_arbitrum', 'InvestmentPoolFactory_call_create') }} cc
-        ON c.evt_tx_hash = cc.call_tx_hash
-        AND SUBSTRING(c.poolId, 0, 42) = cc.output_0
-        {% if is_incremental() %}
-        WHERE c.evt_block_time >= date_trunc("day", now() - interval '1 week')
-          AND cc.call_block_time >= date_trunc("day", now() - interval '1 week')
-        {% endif %}
-    )
-
-    UNION ALL
-
-    SELECT pool_id,
-           zip.tokens AS token_address,
-           zip.weights / pow(10, 18) AS normalized_weight,
-           symbol,
-           pool_type
-    FROM (
-        SELECT c.poolId AS pool_id,
-               explode(arrays_zip(cc.tokens, cc.weights)) AS zip,
-               cc.symbol,
-               'WP2T' AS pool_type
-        FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} c
-        INNER JOIN {{ source('balancer_v2_arbitrum', 'WeightedPool2TokensFactory_call_create') }} cc
-        ON c.evt_tx_hash = cc.call_tx_hash
-        AND SUBSTRING(c.poolId, 0, 42) = cc.output_0
-        {% if is_incremental() %}
-        WHERE c.evt_block_time >= date_trunc("day", now() - interval '1 week')
-          AND cc.call_block_time >= date_trunc("day", now() - interval '1 week')
-        {% endif %}
-    )
-
-    UNION ALL
-
-    SELECT c.poolId AS pool_id,
-           explode(cc.tokens) AS token_address,
-           CAST(NULL AS DOUBLE) AS normalized_weight,
-           cc.symbol,
-           'SP' AS pool_type
+  SELECT
+    pool_id,
+    zip.tokens AS token_address,
+    zip.weights / pow(10, 18) AS normalized_weight,
+    symbol,
+    pool_type
+  FROM (
+    SELECT
+      c.poolId AS pool_id,
+      t.tokens,
+      w.weights,
+      cc.symbol,
+      'WP' AS pool_type
     FROM {{ source('balancer_v2_arbitrum', 'Vault_evt_PoolRegistered') }} c
-    INNER JOIN {{ source('balancer_v2_arbitrum', 'StablePoolFactory_call_create') }} cc
-    ON c.evt_tx_hash = cc.call_tx_hash
-    AND SUBSTRING(c.poolId, 0, 42) = cc.output_0
+    INNER JOIN {{ source('balancer_v2_arbitrum', 'WeightedPoolFactory_call_create') }} cc
+      ON c.evt_tx_hash = cc.call_tx_hash
+      AND SUBSTRING(CAST(c.poolId AS varchar), 1, 42) = CAST(cc.output_0 AS varchar)
```

Comment:
Hi @viniabussafi, you can try this for poolId mapping:
```suggestion
      AND bytearray_substring(c.poolId, 1, 20) = cc.output_0
```

Reply:
Thanks for the review @Hosuke! It makes sense. I already made the changes :)